sympy/solvers/bivariate.py

```diff
@@ -179,8 +179,25 @@ def _lambert(eq, x):
     p, den = den.as_coeff_Mul()
     e = exp(num/den)
     t = Dummy('t')
-    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
-
+    if p == 1 :
+        t = e
+        args = [d/(a*b)*t]
+    elif domain.is_subset(S.Reals):
+        args1 = [d/(a*b)*t for t in roots(t**p - e, t).keys() ]
+        args = []
+        j = -1
+        ind_ls = []
+        for i in args1:
+            j += 1
+            de = "{}".format(i)
+            if 'I' not in de:
+                ind_ls.append(j)
+        for i in ind_ls:
+            args.append(args1[i])
+    else:
+        args = [d/(a*b)*t for t in roots(t**p - e, t).keys() ]
+    if len(args) == 0:
+        return S.EmptySet
```

Comment:
Case(2)
Also the argument part as domain is included at _lambert , _solve_lambert , _solve_even_degree_expr to solve in real domain